### ISSUE TITLE
Chrome 44: Use ServiceWorkerRegistration.showNotification() instead.

### ISSUE THREAD

**Issue by zspitzer** [alexgibson/notify.js#38]
Uncaught TypeError: Failed to construct 'Notification': Illegal constructor. Use ServiceWorkerRegistration.showNotification() instead.

**Comment by alexgibson** [comment]
This lib does not yet support service worker type notifications

**Comment by zspitzer** [comment]
in the meantime, could isSupported be updated?
as per
https://developers.google.com/web/updates/2015/05/Notifying-you-of-notificiation-changes?hl=en
 try {
        new Notification('');
    } catch (e) {
        if (e.name == 'TypeError')
            return false;
    }

**Comment by alexgibson** [comment]
Thanks! I'll dig through this

**Comment by chrisyip** [comment]
I got this error on `Chrome Mobile 45.0` on `Android 5.1.1` too.

```
Failed to construct 'Notification': Illegal constructor. Use ServiceWorkerRegistration.showNotification() instead.
```

**Comment by alexgibson** [comment]
I don't have an Android device to test, but this should work hopefully:

https://github.com/alexgibson/notify.js/tree/notification-feature-detect

Can someone please test this branch to see if the error is gone? If so, I'll update master and push a new tag for release.

Thanks all!

**Comment by chrisyip** [comment]
I don't have an Android device too.

I tried to write a demo and ask my colleague to do some tests, but still failing.

Coz we work remotely, I can't directly debug on her device, don't know what's going on.

Sorry for can't provide more informations.

**Comment by alexgibson** [comment]
Thanks - did your colleague try the branch linked above in Comment 17?

**Comment by alexgibson** [comment]
To be clear, this patch won't make Notifications from web pages work on Android. It will just prevent the fade positive occurring when using the isSupported() method. It should be returning false.

**Comment by alexgibson** [comment]
@zspitzer are you able to help test this? It would be much appreciated

**Comment by zspitzer** [comment]
almost works, the problem is that permission is still granted in Chrome so 
you need to test after it's granted
```
        if (Notification.permission === 'granted') {
            // We would only have prompted the user for permission if new
            // Notification was supported, so assume it is supported.

            // https://developers.google.com/web/updates/2015/05/Notifying-you-of-notificiation-changes
            try {
                new Notification('');
            } catch (e) {
                if (e.name === 'TypeError') {
                    return false;
                }
            }
            return true;
        }
```

**Comment by zspitzer** [comment]
alas, in desktop chrome, I'm now seeing a empty notification for each call to Notify.isSupported

**Comment by alexgibson** [comment]
Thanks for the feedback - i'll see what I can work out.

**Comment by alexgibson** [comment]
@zspitzer I can't test for support after permission is granted due to ghost notifications like you point out. I had assumed permission would never be `granted` on Chrome Android since it doesn't support Web Notifications (hence why I'm returning `true` for that condition). Is this not the case? Just trying to figure out what's not working.

**Comment by alexgibson** [comment]
@zspitzer - can you please provide some more info, as I'm not quite clear on what the issue is from your feedback? Can you please clarify when you say "Chrome", if you mean Chrome for Desktop or Chrome for Android?

I'm a bit confused as to how `Notification.permission === 'granted'` could be true on Chrome for Android, if it does not support Web Notifications?

**Comment by zspitzer** [comment]
It previously did but they deprecated it in v44 on Android mobile, hence the odd nature of it being granted (via the previous version) quite a mess

**Comment by alexgibson** [comment]
Ok this makes more sense now, thanks again for the feedback.

**Comment by alexgibson** [comment]
Ok this issue should hopefully be resolved (the best we can) in v1.2.6 https://github.com/alexgibson/notify.js/releases/tag/v1.2.6
